compaction.go

```diff
@@ -541,10 +548,35 @@ func compactLevel0IntoLevel1(
 
 	bufs := make([]dynparquet.DynamicRowGroup, 0, len(level0Parts))
 	for _, p := range partsToCompact {
+		buf := p.Buf()
+
+		// If this is a Arrow record part, convert the record into a serialized buffer
```

Comment:
I wonder if a better interface would be to make this internal to the `parts` package and maybe also make the encoding explicit. i.e.:
```
// Returns whether the part is an Arrow record part.
(p *Part) IsArrow() bool

etc...
// AsSerializedBuffer returns the data as a parquet serialized buffer, if this
// is an arrow record part, the record will be converted to a serialized buffer.
(p *Part) AsSerializedBuffer()
```

Reply:
I think I also saw another case below that would benefit from this `AsSerializedBuffer`

Reply:
Great ideas! Will do